shell/core/src/main/java/org/apache/karaf/shell/impl/console/ConsoleSessionImpl.java

```diff
@@ -299,7 +299,9 @@ public Object evaluate(LineReader reader, ParsedLine line, String func) throws E
      * @return the history file
      */
     protected Path getHistoryFile() {
-        String defaultHistoryPath = new File(System.getProperty("user.home"), ".karaf/karaf.history").toString();
+        String instanceName = System.getProperty("karaf.name", "root");
```

Comment:
Might want to keep `karaf.history` filename for the root instance for retrocompatibility? If so there needs to be a condition here.

Reply:
Yes, if instance name is `root`, it would be great to keep `karaf.history` for the name (for backward compatibility).

Reply:
Done, it will stay `karaf.history` now!